src/NuGet.Clients/NuGet.PackageManagement.UI/Actions/UIActionEngine.cs

```diff
@@ -342,16 +342,20 @@ public sealed class UIActionEngine
 #else
             var potentialProjects = new List<NuGetProject>();
 
-            // check if project is packages.config and it's dte project instance can also be converted to VSProject4.
+            // check if project suppports <PackageReference> items.
             // otherwise don't show format selector dialog for this project
-            var msBuildProjects = uiService.Projects.Where(project =>
-                project is MSBuildNuGetProject &&
-                (project as MSBuildNuGetProject).MSBuildNuGetProjectSystem.VSProject4 != null);
+            var capableProjects = uiService
+                .Projects
+                .Where(project =>
+                    project.ProjectStyle == ProjectModel.ProjectStyle.PackagesConfig &&
+                    project.ProjectServices.Capabilities.SupportsPackageReferences);
```

Comment:
`MSBuildNuGetProject` initializes its `ProjectStyle` with `ProjectStyle.PackagesConfig` unconditionally. So checking the style here is equivalent to checking the type cast which is less efficient.